pages/flow/[id].js

```diff
@@ -0,0 +1,254 @@
+import { flowDummys } from "../../db";
+import useLocalStorage from "../../hooks/useLocalStorage";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import AsanaCard from "../../components/Asana-Card/AsanaCard";
+import asanas from "../../db";
+import { useState } from "react";
+import { MainButton } from "../../components/MainButton";
+import { nanoid } from "nanoid";
+import SearchBar from "../../components/SearchBar";
+import LevelFilter from "../../components/LevelFilter";
+
+export default function FlowPage() {
+  const [flows, setFlows] = useLocalStorage("flows", flowDummys);
+  const router = useRouter();
+  const { id } = router.query;
+  const [open, setOpen] = useState(false);
+  const [searchQuery, setSearchQuery] = useState("");
+  const [filterQuery, setFilterQuery] = useState("all");
+
+  const currentFlow = flows.find((flow) => flow.id === id);
+  const name = currentFlow?.name || "No flow found";
+  const hours = currentFlow?.duration?.hours || "";
+  const minutes = currentFlow?.duration?.minutes || "";
+  const currentAsanas = currentFlow?.asanas || [];
+
+  function updateFlow(id, asana) {
+    setFlows(
+      flows.map((flow) => {
+        if (flow.id == id) {
+          return {
+            ...flow,
+            asanas: [...flow.asanas, { ...asana, flowListId: nanoid() }],
+          };
+        } else return flow;
+      })
+    );
+  }
+
+  function resetFlow(id) {
+    setFlows(
+      flows.map((flow) => {
+        if (flow.id == id) {
+          return {
+            ...flow,
+            asanas: [],
+          };
+        } else return flow;
+      })
+    );
+  }
+
+  function deleteAsana(cardID) {
+    const filteredAsanas = currentFlow.asanas.filter(
+      (asana) => asana.flowListId !== cardID
+    );
+    setFlows(
+      flows.map((flow) => {
+        if (flow.id == id) {
+          return {
+            ...flow,
+            asanas: [...filteredAsanas],
```

Comment:
There is actually no need to spread this array here, since it is already a new object created in line 54.

Reply:
Ah, true! :)